CONTRIBUTING.md

```diff
@@ -11,14 +21,39 @@ You should be able to run `make init` to install the dependencies and then `make
 
 ## Linting
 
-Run `make lint` or `black --check moto tests` to verify whether your code confirms to the guidelines.
-
-## Getting to grips with the codebase
+Ensure that the correct version of black is installed - `black==19.10b0`. (Different versions of black will return different results.)  
+Run `make lint` to verify whether your code confirms to the guidelines.  
+Use `make format` to automatically format your code, if it does not conform to `black`'s rules.
 
-Moto maintains a list of [good first issues](https://github.com/spulec/moto/contribute) which you may want to look at before
-implementing a whole new endpoint.
+## General notes
 
-## Missing features
+Some tips that might help during development:
+ - A dedicated TaggingService exists in `moto.utilities`, to help with storing/retrieving tags for resources. Not all services use it yet, but contributors are encouraged to  use the TaggingService for all new features.  
+ - Our CI runs all tests twice - one normal run, and one run in ServerMode. In ServerMode, Moto is started as a stand-alone Flask server, and all tests are run against this Flask-instance.
+To verify whether your tests pass in ServerMode, you can run the following commands:
+```
+pip install .[all,server]
+moto_server
+TEST_SERVER_MODE=true pytest -sv tests/test_service/..
+```
+ - When writing tests, one test should only verify a single feature/method. I.e., one test for `create_resource()`, another for `update_resource()`, etc.
+ - When writing negative tests, try to use the following format:
+ ```
+ with pytest.raises(botocore.exceptions.ClientError) as ex:
+     client.failing_call(..)
+ err = ex.value.response["Error"]
+ err["Code"].should.equal(..)
+ err["Message"].should.equal(..)
+ ```
```

Comment:
pylint expects variables to be at least three characters long, so could you change "ex" to "exc" in the example above?

Reply:
Would you want to allow the possibility of using assert() as well as `sure`?  

It might be worth noting that the above code is using methods from the `sure` library ... I hadn't seen or used that library before this and if I saw that code, I might be a little confused by it.

Reply:
Done and done